### ISSUE TITLE
Build fail on macOS Sierra 10.12.6

### ISSUE THREAD

**Issue by vadimmelnicuk** [MultiChain/multichain#34]
After following the procedure for mac, I got a failed build:

```
Making all in src
  CXX      multichain/multichaind-multichaind.o
  CXX      rpc/multichaind-rpclist.o
  CXX      chainparams/multichaind-buildgenesis.o
  CXX      storage/libbitcoin_server_a-addrman.o
  CXX      structs/libbitcoin_server_a-alert.o
  CXX      structs/libbitcoin_server_a-bloom.o
  CXX      chain/libbitcoin_server_a-chain.o
  CXX      chain/libbitcoin_server_a-checkpoints.o
  CXX      core/libbitcoin_server_a-init.o
  CXX      storage/libbitcoin_server_a-leveldbwrapper.o
  CXX      core/libbitcoin_server_a-main.o
  CXX      protocol/libbitcoin_server_a-multichaintx.o
protocol/multichaintx.cpp:855:37: warning: unused variable 'pc1' [-Wunused-variable]
            CScript::const_iterator pc1 = script1.begin();
                                    ^
protocol/multichaintx.cpp:2052:37: warning: unused variable 'pc1' [-Wunused-variable]
            CScript::const_iterator pc1 = script1.begin();
                                    ^
2 warnings generated.
  CXX      protocol/libbitcoin_server_a-multichainblock.o
  CXX      protocol/libbitcoin_server_a-handshake.o
  CXX      chain/libbitcoin_server_a-merkleblock.o
  CXX      miner/libbitcoin_server_a-miner.o
  CXX      net/libbitcoin_server_a-net.o
  CXX      ui/libbitcoin_server_a-noui.o
  CXX      chain/libbitcoin_server_a-pow.o
  CXX      net/libbitcoin_server_a-rest.o
  CXX      utils/libbitcoin_server_a-utilparse.o
  CXX      json/libbitcoin_server_a-json_spirit_ubjson.o
In file included from json/json_spirit_ubjson.cpp:4:
./json/json_spirit_ubjson.h:19:7: warning: 'ubjson_read' has C-linkage specified, but returns incomplete type 'Value' (aka 'Value_impl<json_spirit::Config_vector<std::__1::basic_string<char> > >') which could be
      incompatible with C [-Wreturn-type-c-linkage]
Value ubjson_read(const unsigned char *elem,size_t elem_size,int max_depth,int *err);
      ^
json/json_spirit_ubjson.cpp:358:39: error: conversion from 'size_type' (aka 'unsigned long') to 'const json_spirit::Value_impl<json_spirit::Config_vector<std::__1::basic_string<char> > >' is ambiguous
                    value=array_value.size();
                          ~~~~~~~~~~~~^~~~~~
./json/json_spirit_value.h:46:9: note: candidate constructor
        Value_impl( bool               value );
        ^
./json/json_spirit_value.h:47:9: note: candidate constructor
        Value_impl( int                value );
        ^
./json/json_spirit_value.h:48:9: note: candidate constructor
        Value_impl( int64_t     value );
        ^
./json/json_spirit_value.h:49:9: note: candidate constructor
        Value_impl( uint64_t    value );
        ^
./json/json_spirit_value.h:50:9: note: candidate constructor
        Value_impl( double             value );
        ^
json/json_spirit_ubjson.cpp:446:37: error: conversion from 'size_type' (aka 'unsigned long') to 'const json_spirit::Value_impl<json_spirit::Config_vector<std::__1::basic_string<char> > >' is ambiguous
                    value=obj_value.size();
                          ~~~~~~~~~~^~~~~~
./json/json_spirit_value.h:46:9: note: candidate constructor
        Value_impl( bool               value );
        ^
./json/json_spirit_value.h:47:9: note: candidate constructor
        Value_impl( int                value );
        ^
./json/json_spirit_value.h:48:9: note: candidate constructor
        Value_impl( int64_t     value );
        ^
./json/json_spirit_value.h:49:9: note: candidate constructor
        Value_impl( uint64_t    value );
        ^
./json/json_spirit_value.h:50:9: note: candidate constructor
        Value_impl( double             value );
        ^
1 warning and 2 errors generated.
make[1]: *** [json/libbitcoin_server_a-json_spirit_ubjson.o] Error 1
make: *** [all-recursive] Error 1
```

**Comment by gidgreen** [comment]
We're aware of this and hope to fix it shortly.

**Comment by gidgreen** [comment]
We've released a `1.0.x-mac-fix` branch that hopefully fixes this – please let us know: https://github.com/MultiChain/multichain/tree/1.0.x-mac-fix
